### PR TITLE
API: Add keyword for match_node_purpose

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pangeo-data/pangeo-stacks/issues/59

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @jhamman @scottyhq

**Comment by jacobtomlinson** [comment]
Looks like this has some merge conflicts after the big refactor in #162. @TomAugspurger are you ok to resolve those?

Sorry!

**Comment by scottyhq** [comment]
I think updating or merging this would be helpful (see https://github.com/pangeo-data/pangeo-cloud-federation/pull/567). @TomAugspurger can you give an example of how users can select this setting in dask_cofig.yaml or elsewhere? 

For example in JupyterHub values.yaml we can set:
```
  jupyterhub:
    scheduling:
      userPods:
        nodeAffinity:
          matchNodePurpose: prefer
      corePods:
        nodeAffinity:
          matchNodePurpose: require
```

**Comment by TomAugspurger** [comment]
Hopefully got the rebase correct. Haven't run the tests or anything locally
though.

**Comment by jacobtomlinson** [comment]
Thanks @TomAugspurger. Not sure what worked as CI is failing.

**Comment by jacobtomlinson** [comment]
Thanks @TomAugspurger. CI seems to be failing now. Could you take a look?

File: dask_kubernetes/kubernetes.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ kubernetes:
   host: "0.0.0.0"
   port: 0
   env: {}
+  match_node_purpose: "prefer"
```
Comment (jsignell) [inline dask_kubernetes/kubernetes.yaml:10]: I think this would be better as `match-node-purpose`

**Comment by jacobtomlinson** [comment]
I am going to close this out as it is rather old and I think the Pangeo folks have moved to `dask-gateway anyway.

Sorry this got dropped. Feel free to ping here if it still has value.

**Pull request closed without merging**